src/bridge/cockpitcpusamples.c

```diff
@@ -71,28 +71,45 @@ cockpit_cpu_samples (CockpitSamples *samples)
       guint64 system;
       guint64 idle;
       guint64 iowait;
+      gchar cpu_core[8];
+      gchar core[5];
 
-      if (!(g_str_has_prefix (line, "cpu ")))
+      if (!(g_str_has_prefix (line, "cpu")))
         continue;
 
       #define FMT64 "%" G_GUINT64_FORMAT " "
-      if (sscanf (line + sizeof ("cpu ") - 1, FMT64 FMT64 FMT64 FMT64 FMT64,
+      if (sscanf (line, "%s " FMT64 FMT64 FMT64 FMT64 FMT64,
+                  cpu_core,
                   &user,
                   &nice,
                   &system,
                   &idle,
-                  &iowait) != 5)
+                  &iowait) != 6)
         {
           g_warning ("Error parsing line %d of /proc/stat with content `%s'", n, line);
           continue;
         }
 
       user_hz = ensure_user_hz ();
-      cockpit_samples_sample (samples, "cpu.basic.nice", NULL, nice*1000/user_hz);
-      cockpit_samples_sample (samples, "cpu.basic.user", NULL, user*1000/user_hz);
-      cockpit_samples_sample (samples, "cpu.basic.system", NULL, system*1000/user_hz);
-      cockpit_samples_sample (samples, "cpu.basic.iowait", NULL, iowait*1000/user_hz);
-      break;
+      if (strlen (cpu_core) > 3)
+        {
+          if (sscanf (cpu_core, "cpu%s", core) != 1)
```

Comment:
You could keep the "cpu" prefix in the initial sscanf(), then you only have to deal with "empty string" or "a string number", and can drop the "3". Then you can case-differentiate with `!cpu_core[0]`. Then  this sscanf reduces to `atoi(cpu_core)`.

Reply:
yeah, that was my initial idea but I was not able to write format string that would accept "possibly empty string".
The format is either "cpu  1 2 3" or "cpu14 1 2 3" and anything I came up with has then eaten the first value in the first case as core number. Any advice is welcomed :)

Reply:
I see -- let's keep it as-is then. Thanks!